### ISSUE TITLE
Incorrect Html content parsing

### ISSUE THREAD

**Issue by pytnik89** [zbateson/MailMimeParser#38]
Hello, my friend.
First of all let me thanks you for your awesome lib. It's a realy good alternative to those that uses mailparse extension.
I found some strange behaviour when parse simple mail content.
When I just echo html content it differs from that in email.
`$email = file_get_contents($_SERVER['DOCUMENT_ROOT'].'/mail_debug_content.txt');
$mailParser = new ZBateson\MailMimeParser\MailMimeParser();
$Message = $mailParser->parse($email);
$html = $Message->getHtmlContent();
$atts = $Message->getAllAttachmentParts();
echo $html;
`
If it is due to of my misunderstanding of what's going on, please give advise to where look up.

Thanks a lot for your help!
 
[mail_debug_content.txt](https://github.com/zbateson/MailMimeParser/files/878313/mail_debug_content.txt)

**Comment by zbateson** [comment]
Hi @pytnik89 -

Thanks for the lovely feedback :)

The code you pasted all looks good to me... could you tell me how the output differs?  My best guess is it could be a character set issue if you're sending back as a response to a web request (could be affected by setting mb_http_output to something other than UTF-8, or by the character set in the output if it's not UTF-8, e.g. in header('Content-Type: text/html; charset=something-else') or in a meta tag.)

All the best,
Zaahid

**Comment by pytnik89** [comment]
Thx a lot for your replay, yes you are right this behaviour connected with charset. In file \Stream\PartStream.php at line 154 there is are definition $this->position += strlen($ret), and this value used in fread as a second parameter, but fread expects byte position not chars counts, in my message one char is two bytes this why I have a problem. Thx again for your help I will try to rewrite this part of code.

**Comment by zbateson** [comment]
Hi @pytnik89 --

That's awesome, thanks for investigating!

I had a quick try, and reading your email seemed to work for me.  strlen should return byte count anyway (mb_strlen returns character count).

**Comment by pytnik89** [comment]
Thanks my friend, yes you are right, that strlen should return byte count, but not in my case... It's a realy often case when in some specific language projects with a lot of legacy, uses mbstring.func_overload for compatible with multibyte...so in my case there is mbstirng.func_overload=2. In this case it's may be better to use ini_get('mbstring.func_overload') ? mb_strlen($string, '8bit') : strlen($string), because we cant to trust strlen any more.

**Comment by zbateson** [comment]
Hi @pytnik89 -

I think MailMimeParser is a good candidate for this solution.  Headers should be ASCII anyway before conversion, so I think it would be enough to just change the one 'strlen' call in PartStream.

If you implement it in a really cool way and submit a pull request I'll merge it :smiling_imp: I would prefer ini_get only be called once, perhaps setting a static variable in PartStream.

Otherwise I'll try get to it in the next couple of weeks.  Looking around it doesn't seem like a big deal to implement.  These guys were kind of concerned about speed, but I think for this library that's not as much of a concern given what we're doing and where we're using it:

https://github.com/phpseclib/phpseclib/issues/762

So in short, I accept the idea.

All the best,
Zaahid
